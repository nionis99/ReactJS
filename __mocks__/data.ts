export const genres = ['Crime', 'Documentary', 'Horror', 'Comedy'];
export const MovieMockData = {
  id: 0,
  title: 'The Avengers 2',
  vote_average: 7.5,
  release_date: '2012-04-25',
  poster_path: 'https://image.tmdb.org/t/p/w500/cezWGskPY5x7GaglTTRN4Fugfb8.jpg',
  overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
  genres: ['Science Fiction', 'Action', 'Adventure'],
  runtime: 143,
};
