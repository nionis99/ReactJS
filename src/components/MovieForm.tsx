import React from 'react';
import Input from 'components/Input';
import SelectInput from 'components/SelectInput';
import Button from 'components/Button';
import { genres } from '../../__mocks__/data';

interface MovieFormProps {
  onSubmit: () => void;
  onReset: () => void;
}

const MovieForm = ({ onSubmit, onReset }: MovieFormProps) => {
  return (
    <form className="w-full" onSubmit={onSubmit}>
      <div className="flex flex-wrap -mx-3 mb-4">
        <Input className="w-full sm:w-2/3 px-3 mb-6 md:mb-0" label="Title" type="text" placeholder="Movie title" />
        <Input className="w-full sm:w-1/3 px-3" label="Release date" type="date" placeholder="Select Date" />
      </div>
      <div className="flex flex-wrap -mx-3 mb-4">
        <Input className="w-full sm:w-2/3 px-3 mb-6 md:mb-0" label="Movie url" type="text" placeholder="https://" />
        <Input
          className="w-full sm:w-1/3 px-3"
          label="Rating"
          type="number"
          min="0"
          max="10"
          step="0.1"
          placeholder="7.8"
        />
      </div>
      <div className="flex flex-wrap -mx-3 mb-4">
        <SelectInput className="w-full sm:w-2/3 px-3 mb-6 md:mb-0" label="Genre" options={genres} />
        <Input
          className="w-full sm:w-1/3 px-3 mb-6 md:mb-0"
          label="runtime"
          placeholder="minutes"
          type="number"
          step="1"
          min="1"
        />
      </div>
      <div className="flex flex-wrap -mx-3 mb-4">
        <div className="w-full px-3 mb-6 md:mb-0">
          <label className="block uppercase tracking-wide text-xs font-bold mb-2 upper-case text-primary">
            Overview
          </label>
          <textarea
            className="appearance-none w-full bg-gray80 border-gray80 text-white border rounded py-2 px-4 mb-3 
            leading-tight focus:outline-none focus:border-gray-200"
            name="overview"
            placeholder="Movie description"
            rows={6}
          />
        </div>
      </div>
      <div className="flex flex-row w-full">
        <Button title="Reset" className="ml-auto mr-4" variant="outline-primary" size="medium" onClick={onReset} />
        <Button title="Submit" variant="primary" size="medium" type="submit" />
      </div>
    </form>
  );
};

export default MovieForm;
