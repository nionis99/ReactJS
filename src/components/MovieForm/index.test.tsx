import React from 'react';
import renderer from 'react-test-renderer';
import MovieForm from '.';
import { MovieMockData } from '../../../__mocks__/data';

describe('Movie form', () => {
  const onSubmit = jest.fn();

  it('should render movie form w/o data with loading button', () => {
    const elem = renderer.create(<MovieForm isLoading={true} onSubmit={onSubmit} />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should render movie form with filled data', () => {
    const elem = renderer.create(<MovieForm movie={MovieMockData} isLoading={false} onSubmit={onSubmit} />).toJSON();
    expect(elem).toMatchSnapshot();
  });
});
