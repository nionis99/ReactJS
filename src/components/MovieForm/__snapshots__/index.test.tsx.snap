// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie form should render movie form w/o data with loading button 1`] = `
<form
  className="w-full"
  onSubmit={[Function]}
>
  <div
    className="flex flex-wrap -mx-3 mb-4"
  >
    <div
      className="w-full sm:w-2/3 px-3 mb-6 md:mb-0"
    >
      <label
        className="block tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        Title
      </label>
      <input
        className="appearance-none block w-full bg-gray80 text-white border rounded py-2 h-12 px-4 mb-3 leading-tight focus:outline-none focus:border-gray-200 border-gray80"
        defaultValue=""
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Movie title"
        type="text"
      />
    </div>
    <div
      className="w-full sm:w-1/3 px-3"
    >
      <label
        className="block tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        Release date
      </label>
      <input
        className="appearance-none block w-full bg-gray80 text-white border rounded py-2 h-12 px-4 mb-3 leading-tight focus:outline-none focus:border-gray-200 border-gray80"
        name="release_date"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Select Date"
        type="date"
      />
    </div>
  </div>
  <div
    className="flex flex-wrap -mx-3 mb-4"
  >
    <div
      className="w-full sm:w-2/3 px-3 mb-6 md:mb-0"
    >
      <label
        className="block tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        Movie url
      </label>
      <input
        className="appearance-none block w-full bg-gray80 text-white border rounded py-2 h-12 px-4 mb-3 leading-tight focus:outline-none focus:border-gray-200 border-gray80"
        name="poster_path"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="https://"
        type="text"
      />
    </div>
    <div
      className="w-full sm:w-1/3 px-3"
    >
      <label
        className="block tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        Rating
      </label>
      <input
        className="appearance-none block w-full bg-gray80 text-white border rounded py-2 h-12 px-4 mb-3 leading-tight focus:outline-none focus:border-gray-200 border-gray80"
        max="10"
        min="0"
        name="vote_average"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="7.8"
        step="0.1"
        type="number"
      />
    </div>
  </div>
  <div
    className="flex flex-wrap -mx-3 mb-4"
  >
    <div
      className="w-full sm:w-2/3 px-3 mb-6 md:mb-0"
    >
      <label
        className="block uppercase tracking-wide text-primary text-xs font-bold mb-2"
      >
        Genre
      </label>
      <div
        className="flex flex-row relative"
      >
        <div
          className="flex flex-row appearance-none w-full bg-gray80 border border-gray80 text-white h-12 py-2 px-4 rounded leading-tight focus:outline-none focus:border-gray-200 cursor-pointer"
          data-testid={8}
          onClick={[Function]}
        >
          <span
            className="flex items-center justify-start select-none"
          >
            Select genre
          </span>
          <div
            className="flex ml-auto items-center"
          >
            <test-file-stub
              className="flex justify-end items-center fill-current h-3 w-3"
            />
          </div>
          <ul
            className="absolute w-full right-0 top-10 py-1 mt-2 bg-white bg-content z-2 shadow-modal select-none hidden"
          >
            <li
              className="flex flex-row text-white w-full"
            >
              <input
                checked={false}
                className="m-auto"
                id="Crime"
                name="Crime"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="flex ml-2 px-4 py-2 w-full cursor-pointer"
                htmlFor="Crime"
              >
                Crime
              </label>
            </li>
            <li
              className="flex flex-row text-white w-full"
            >
              <input
                checked={false}
                className="m-auto"
                id="Documentary"
                name="Documentary"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="flex ml-2 px-4 py-2 w-full cursor-pointer"
                htmlFor="Documentary"
              >
                Documentary
              </label>
            </li>
            <li
              className="flex flex-row text-white w-full"
            >
              <input
                checked={false}
                className="m-auto"
                id="Horror"
                name="Horror"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="flex ml-2 px-4 py-2 w-full cursor-pointer"
                htmlFor="Horror"
              >
                Horror
              </label>
            </li>
            <li
              className="flex flex-row text-white w-full"
            >
              <input
                checked={false}
                className="m-auto"
                id="Comedy"
                name="Comedy"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="flex ml-2 px-4 py-2 w-full cursor-pointer"
                htmlFor="Comedy"
              >
                Comedy
              </label>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="w-full sm:w-1/3 px-3 mb-6 md:mb-0"
    >
      <label
        className="block tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        runtime
      </label>
      <input
        className="appearance-none block w-full bg-gray80 text-white border rounded py-2 h-12 px-4 mb-3 leading-tight focus:outline-none focus:border-gray-200 border-gray80"
        min="1"
        name="runtime"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="minutes"
        step="1"
        type="number"
      />
    </div>
  </div>
  <div
    className="flex flex-wrap -mx-3 mb-4"
  >
    <div
      className="w-full px-3 mb-6 md:mb-0"
    >
      <label
        className="block uppercase tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        Overview
      </label>
      <textarea
        className="appearance-none w-full bg-gray80  border-gray80 text-white border rounded py-2 px-4 mb-3
            leading-tight focus:outline-none focus:border-gray-200 resize-none"
        data-testid={9}
        name="overview"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Movie description"
        rows={6}
      />
    </div>
  </div>
  <div
    className="flex flex-row w-full"
  >
    <button
      className="flex justify-center items-center rounded border p-2 border-primary text-primary px-2 md:px-11 py-1 md:py-2.5 ml-auto mr-4"
      onClick={[Function]}
      type="reset"
    >
      Reset
    </button>
    <button
      className="flex justify-center items-center rounded border p-2 text-white bg-primary border-primary px-2 md:px-11 py-1 md:py-2.5"
      data-testid={10}
    >
      <div
        className="flex flex-grow h-full justify-center items-center"
      >
        <div
          className="animate-spin rounded-full border-b-2 border-white w-6 h-6"
        />
      </div>
    </button>
  </div>
</form>
`;

exports[`Movie form should render movie form with filled data 1`] = `
<form
  className="w-full"
  onSubmit={[Function]}
>
  <div
    className="flex flex-wrap -mx-3 mb-4"
  >
    <div
      className="w-full sm:w-2/3 px-3 mb-6 md:mb-0"
    >
      <label
        className="block tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        Title
      </label>
      <input
        className="appearance-none block w-full bg-gray80 text-white border rounded py-2 h-12 px-4 mb-3 leading-tight focus:outline-none focus:border-gray-200 border-gray80"
        defaultValue="The Avengers 2"
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Movie title"
        type="text"
      />
    </div>
    <div
      className="w-full sm:w-1/3 px-3"
    >
      <label
        className="block tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        Release date
      </label>
      <input
        className="appearance-none block w-full bg-gray80 text-white border rounded py-2 h-12 px-4 mb-3 leading-tight focus:outline-none focus:border-gray-200 border-gray80"
        defaultValue="2012-04-25"
        name="release_date"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Select Date"
        type="date"
      />
    </div>
  </div>
  <div
    className="flex flex-wrap -mx-3 mb-4"
  >
    <div
      className="w-full sm:w-2/3 px-3 mb-6 md:mb-0"
    >
      <label
        className="block tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        Movie url
      </label>
      <input
        className="appearance-none block w-full bg-gray80 text-white border rounded py-2 h-12 px-4 mb-3 leading-tight focus:outline-none focus:border-gray-200 border-gray80"
        defaultValue="https://image.tmdb.org/t/p/w500/cezWGskPY5x7GaglTTRN4Fugfb8.jpg"
        name="poster_path"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="https://"
        type="text"
      />
    </div>
    <div
      className="w-full sm:w-1/3 px-3"
    >
      <label
        className="block tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        Rating
      </label>
      <input
        className="appearance-none block w-full bg-gray80 text-white border rounded py-2 h-12 px-4 mb-3 leading-tight focus:outline-none focus:border-gray-200 border-gray80"
        defaultValue={7.5}
        max="10"
        min="0"
        name="vote_average"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="7.8"
        step="0.1"
        type="number"
      />
    </div>
  </div>
  <div
    className="flex flex-wrap -mx-3 mb-4"
  >
    <div
      className="w-full sm:w-2/3 px-3 mb-6 md:mb-0"
    >
      <label
        className="block uppercase tracking-wide text-primary text-xs font-bold mb-2"
      >
        Genre
      </label>
      <div
        className="flex flex-row relative"
      >
        <div
          className="flex flex-row appearance-none w-full bg-gray80 border border-gray80 text-white h-12 py-2 px-4 rounded leading-tight focus:outline-none focus:border-gray-200 cursor-pointer"
          data-testid={8}
          onClick={[Function]}
        >
          <span
            className="flex items-center justify-start select-none"
          >
            Science Fiction, Action, Adventure
          </span>
          <div
            className="flex ml-auto items-center"
          >
            <test-file-stub
              className="flex justify-end items-center fill-current h-3 w-3"
            />
          </div>
          <ul
            className="absolute w-full right-0 top-10 py-1 mt-2 bg-white bg-content z-2 shadow-modal select-none hidden"
          >
            <li
              className="flex flex-row text-white w-full"
            >
              <input
                checked={false}
                className="m-auto"
                id="Crime"
                name="Crime"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="flex ml-2 px-4 py-2 w-full cursor-pointer"
                htmlFor="Crime"
              >
                Crime
              </label>
            </li>
            <li
              className="flex flex-row text-white w-full"
            >
              <input
                checked={false}
                className="m-auto"
                id="Documentary"
                name="Documentary"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="flex ml-2 px-4 py-2 w-full cursor-pointer"
                htmlFor="Documentary"
              >
                Documentary
              </label>
            </li>
            <li
              className="flex flex-row text-white w-full"
            >
              <input
                checked={false}
                className="m-auto"
                id="Horror"
                name="Horror"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="flex ml-2 px-4 py-2 w-full cursor-pointer"
                htmlFor="Horror"
              >
                Horror
              </label>
            </li>
            <li
              className="flex flex-row text-white w-full"
            >
              <input
                checked={false}
                className="m-auto"
                id="Comedy"
                name="Comedy"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="flex ml-2 px-4 py-2 w-full cursor-pointer"
                htmlFor="Comedy"
              >
                Comedy
              </label>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="w-full sm:w-1/3 px-3 mb-6 md:mb-0"
    >
      <label
        className="block tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        runtime
      </label>
      <input
        className="appearance-none block w-full bg-gray80 text-white border rounded py-2 h-12 px-4 mb-3 leading-tight focus:outline-none focus:border-gray-200 border-gray80"
        defaultValue={143}
        min="1"
        name="runtime"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="minutes"
        step="1"
        type="number"
      />
    </div>
  </div>
  <div
    className="flex flex-wrap -mx-3 mb-4"
  >
    <div
      className="w-full px-3 mb-6 md:mb-0"
    >
      <label
        className="block uppercase tracking-wide text-xs font-bold mb-2 upper-case text-primary"
      >
        Overview
      </label>
      <textarea
        className="appearance-none w-full bg-gray80  border-gray80 text-white border rounded py-2 px-4 mb-3
            leading-tight focus:outline-none focus:border-gray-200 resize-none"
        data-testid={9}
        defaultValue="Lorem Ipsum is simply dummy text of the printing and typesetting industry."
        name="overview"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Movie description"
        rows={6}
      />
    </div>
  </div>
  <div
    className="flex flex-row w-full"
  >
    <button
      className="flex justify-center items-center rounded border p-2 border-primary text-primary px-2 md:px-11 py-1 md:py-2.5 ml-auto mr-4"
      onClick={[Function]}
      type="reset"
    >
      Reset
    </button>
    <button
      className="flex justify-center items-center rounded border p-2 text-white bg-primary border-primary px-2 md:px-11 py-1 md:py-2.5"
      data-testid={10}
    >
      Edit
    </button>
  </div>
</form>
`;
